api/mod/BallotGeneration.js

```diff
@@ -0,0 +1,195 @@
+const _ = require('lodash');
+
+
+const getfaceOffData = (conn) =>{
+    return conn('Faceoffs AS F')
+    .join('Squads AS S', 'F.SquadID', 'S.ID')
+    .select([
+        'F.ID',
+        'F.SubmissionID1',
+        'F.SubmissionID2',
+        'F.SquadID',
+        'F.VotesCasted',
+        'S.CohortID',
+    ]);
+};
+
+const getChildData = (conn) =>{
+    return conn('Children as C')
+    .select([
+        'C.ID',
+        'C.Ballots',
+    ]);
+};
+
+
+const groupBySquad = (faceoffs) =>{
+    const squads = {}
+    faceoffs.forEach( fo => {
+        if(!squads[fo.SquadID]) squads[fo.SquadID] = []
+        squads[fo.SquadID].push(fo)
+    });
+    return squads
+}
+
+const VSequence = (squads, data) =>{
+
+    let test = null
+    let nsquad = []
+    nsquad.push(squads)
+    let k = Object.keys(squads)
+
+    let newArr = []
+    k.forEach(squad =>{
+        newArr.push(parseInt(squad, 10));
+    })
+
+    let children = filterChildren(data, newArr)
+    let ballots = {}
+    children.forEach(child =>{
+        let num = child.SquadID
+        let votesAvailable = 3;
+        while(votesAvailable > 0){
+            let n = nsquad[nsquad.length-1]
+            for(let squadNum in n){
+                if(squadNum != child.SquadID){
+                    let qualified = leastVotes(n[squadNum], ballots, child.ID)
+                    let choice = getRandomInt(qualified.length)
+                    if(!ballots[child.ID]){
+                        ballots[child.ID] = [qualified[choice].ID]
+                    }else{
+                        ballots[child.ID].push(qualified[choice].ID)
+                    }
+                    nsquad.push(incrementVotesCasted(n, squadNum, [qualified[choice].ID]))
+                    votesAvailable--
+                }
+            }            
+        }    
+    })
+    return ballots
+}
+
+
+const incrementVotesCasted =(squads, ID, FO) =>{
+    let iSquads = squads[`${ID}`]
+    let newArr = []
+    let returnObject = {}
+    iSquads.forEach(s =>{
+        if(s.ID == FO){
+            let newVoteCount = s.VotesCasted + 1
+            newObject = {
+                ID: FO[0],
+                SubmissionID1: s.SubmissionID1,
+                SubmissionID2: s.SubmissionID2,
+                SquadID: s.SquadID,
+                VotesCasted: newVoteCount,
+                CohortID: s.CohortID
+            }
+            newArr.push(newObject)
+        }else{
+            newArr.push(s)
+        }
+
+    for(let squadID in squads){
+        if( squadID == ID){
+            if(!returnObject[squadID]) returnObject[squadID] = newArr
+        }else{
+            if(!returnObject[squadID]) returnObject[squadID] = squads[squadID]
+        }
+    }
```

Comment:
It probably is a neatpick, but it seems to me that you can put the if..else statement inside one `if(!returnObject[squadID])` for a shorter and more readable code. But overall, the code look good to me.